packages/backend-function/src/function_data_config_generator.ts

```diff
@@ -0,0 +1,45 @@
+import fs from 'fs';
+import path from 'path';
+
+const lambdaDataConfigTemplate = (
+  functionName: string
+) => `// This file is auto-generated by Amplify. Edits will be overwritten.
+import { getAmplifyClientsConfiguration } from "@aws-amplify/backend-function/runtime";
+import { env } from "../env/${functionName}";
+export const { libraryOptions, resourceConfig } = await getAmplifyClientsConfiguration(env);
+`;
+
+/**
+ * Generates the data configuration imports
+ */
+export class FunctionDataConfigGenerator {
+  private typeDefFilePath: string;
+
+  private indentation: string = '  ';
+
+  /**
+   * Initialize data configuration file name and location
+   */
+  constructor(private readonly functionName: string) {
+    this.typeDefFilePath = `${process.cwd()}/.amplify/generated/data-config/${
+      this.functionName
+    }.ts`;
+  }
```

Comment:
```suggestion
/**
 * Generates the data configuration imports
 */
export class FunctionAmplifyClientConfigGenerator {
  private typeDefFilePath: string;

  private indentation: string = '  ';

  /**
   * Initialize data configuration file name and location
   */
  constructor(private readonly functionName: string) {
    this.typeDefFilePath = `${process.cwd()}/.amplify/generated/amplify-client-config/${
      this.functionName
    }.ts`;
  }
```

It seems we're doing all of this to call `Amplify.configure` . The runtime objects this generates could be later extended to support more properties (just generating full amplify outputs would be most extreme expansion).
Therefore I think we should use a bit more generic naming here.

Reply:
I had that same thought, however the DX I've been working towards with @josefaidt does all the work and top level `await` to prepare a data centric config (where the config offers specific errors if the MIS isn't available). My thinking is to namespace this tuned experience and if we offer a config function that behaves more generally, that we would offer that on a different import path (maybe the proposed path) and deprecate the data specific experience.

Working ahead of this would assume design work that hasn't happened yet and introduce an extra DX step to get the config from a function that Josef would prefer we avoid.

Reply:
Ok. If we thought about deprecation strategy then it's fine.

Reply:
In the conversation this morning, we decided to move to a `getAmplifyDataClientConfig` function exposed directly from `backend/function/runtime`. Resolving this with that update.